backend/api/src/update-market.ts

```diff
@@ -1,26 +1,53 @@
 import { APIError, authEndpoint, validate } from 'api/helpers'
 import { getContractSupabase, log } from 'shared/utils'
 import * as admin from 'firebase-admin'
-import { record, z } from 'zod'
-import { track, trackPublicEvent } from 'shared/analytics'
+import { z } from 'zod'
+import { trackPublicEvent } from 'shared/analytics'
 import { throwErrorIfNotMod } from 'shared/helpers/auth'
 import { removeUndefinedProps } from 'common/util/object'
-import { createSupabaseDirectClient } from 'shared/supabase/init'
 import { buildArray } from 'common/util/array'
 import { recordContractEdit } from 'shared/record-contract-edit'
+import { MAX_QUESTION_LENGTH, getDescriptionJson } from 'common/contract'
+import { contentSchema } from 'shared/zod-types'
+import { JSONContent } from '@tiptap/core'
 
 const bodySchema = z
   .object({
     contractId: z.string(),
     visibility: z.enum(['unlisted', 'public']).optional(),
     closeTime: z.number().optional(),
+    question: z.string().min(1).max(MAX_QUESTION_LENGTH).optional(),
+    description: contentSchema.or(z.string()).optional(),
+    descriptionHtml: z.string().optional(),
+    descriptionMarkdown: z.string().optional(),
+    descriptionJson: z.string().optional(),
   })
   .strict()
 
 export const updatemarket = authEndpoint(async (req, auth) => {
-  const { contractId, visibility, closeTime } = validate(bodySchema, req.body)
-  if (!visibility && !closeTime)
-    throw new APIError(400, 'Must provide visibility or closeTime')
+  const {
+    contractId,
+    visibility,
+    closeTime,
+    question,
+    description,
+    descriptionHtml,
+    descriptionJson,
+    descriptionMarkdown,
+  } = validate(bodySchema, req.body)
+  if (
+    !visibility &&
+    !closeTime &&
+    !question &&
+    !description &&
+    !descriptionHtml &&
+    !descriptionMarkdown &&
+    !descriptionJson
+  )
+    throw new APIError(
+      400,
+      'Must provide visibility, closeTime, question, description, descriptionHtml, descriptionMarkdown, or descriptionJson'
+    )
```

Comment:
this API error is extraneous - `validate` actually already throws an error if the api params aren't right

Reply:
Doesn't that only do validation on the individual fields though? Seems like this is checking if any field is set, since otherwise it'd be a no-op.